Formula/elan-init.rb

```diff
@@ -37,20 +35,22 @@ def install
   end
 
   test do
-    system bin/"elan-init", "-y"
+    ENV["ELAN_HOME"] = testpath/".elan"
+
+    system bin/"elan-init", "-y", "--default-toolchain=leanprover/lean4:nightly"
     (testpath/"hello.lean").write <<~EOS
       def id' {α : Type} (x : α) : α := x
 
-      inductive tree (α : Type) : Type
-      | node : α → list tree → tree
+      inductive tree {α : Type} : Type
+      | node : α → List tree → tree
 
-      example (a b : Prop) : a ∧ b -> b ∧ a :=
-      begin
-          intro h, cases h,
-          split, repeat { assumption }
-      end
+      example (a b : Prop) : a ∧ b -> b ∧ a := by
+          intro h; cases h with
+          | intro ha hb => constructor; exact hb; exact ha
     EOS
-    system bin/"lean", testpath/"hello.lean"
-    system bin/"leanpkg", "help"
+    system testpath/".elan/bin/lean", testpath/"hello.lean"
+
+    # Nightly versions of LEAN4 don't necessarily provide `leanpkg`.
+    # system testpath/".elan/bin/leanpkg", "help"
```

Comment:
Not keen on removing this check.

Reply:
After some investigation I found out that `leanpkg` will not ship with LEAN4 in favor of `lake`. I'll replace `leanpkg` with `lake` instead.